Sources/SwiftGodot/Core/VariantCollection.swift

```diff
@@ -8,8 +8,14 @@
 @_implementationOnly import GDExtension
 
 /// This represents a typed array of one of the built-in types from Godot
-public class VariantCollection<Element: VariantStorable>: Collection {
-    var array: GArray
+public class VariantCollection<Element: VariantStorable>: Collection, ExpressibleByArrayLiteral {
+    public typealias ArrayLiteralElement = Element
+    
+    public var array: GArray
```

Comment:
exposed this as public so we can assign and and read from it in mproxy set/gets. not sure if I'm stepping on too many toes here.

I was going to create a struct that I realized was identical or very similar to this VariantCollection.